src/GitHub.App/Models/RepositoryHosts.cs

```diff
@@ -157,11 +157,15 @@ public IRepositoryHost LookupHost([AllowNull] HostAddress address)
                     );
                     if (successful)
                     {
+                        // We need to add the connection before assigning to GitHubHost or
+                        // EnterpriseHost so that anything listening to those will find the
+                        // associated connection.
+                        connectionManager.AddConnection(address, usernameOrEmail);
```

Comment:
But what happens if listening code is expecting to have `GitHubHost/EnterpriseHost` set when they get this notification? Sounds like we have a race condition here.